test/suites/telemetry/telemetry.robot

```diff
@@ -16,106 +17,109 @@ Test Tags           restart    slow
+...                                     telemetry:
+...                                     \ \ status: Enabled
+...                                     \ \ endpoint: ${TELEMETRY_WRITE_ENDPOINT}
+...                                     \ \ proxy: ${PROXY_ENDPOINT}
+${ENABLE_TELEMETRY_TO_PROMETHUS}        SEPARATOR=\n
+...                                     telemetry:
+...                                     \ \ status: Enabled
+...                                     \ \ endpoint: http://${PROMETHEUS_HOST}:${PROMETHEUS_PORT}/api/v1/write
+${JOURNAL_CURSOR}                       ${EMPTY}
+${PULL_SECRET}                          /etc/crio/openshift-pull-secret
+${PULL_SECRET_METRICS}                  /etc/crio/openshift-pull-secret-with-telemetry
+${PULL_SECRET_NO_METRICS}               /etc/crio/openshift-pull-secret-without-telemetry
 
 
 *** Test Cases ***
-MicroShift Reports Metrics To Server
+MicroShift Reports Metrics To Default Server
     [Documentation]    Check MicroShift is able to send metrics to the telemetry server without errors.
-    [Setup]    Setup Telemetry Configuration
+    [Setup]    Setup Telemetry Configuration    ${ENABLE_TELEMETRY}    ${PULL_SECRET_METRICS}
 
-    Wait Until Keyword Succeeds    10x    10s
-    ...    Should Find Metrics Success
+    Should Find Metrics In Journal Log Success    MicroShift telemetry starting, sending first metrics collection.
+    Should Find Metrics In Journal Log Success    Metrics sent successfully
 
     [Teardown]    Remove Telemetry Configuration
 
-MicroShift Reports Metrics To Server Through Proxy
+MicroShift Reports Metrics To Default Server Through Proxy
     [Documentation]    Check MicroShift is able to send metrics to the telemetry server through a proxy without errors.
-    [Setup]    Setup Telemetry Configuration With Proxy
-
-    Wait Until Keyword Succeeds    10x    10s
-    ...    Should Find Metrics Success
+    [Setup]    Run Keywords
+    ...    Start Proxy Server    host=${PROXY_HOST}    port=${PROXY_PORT}
+    ...    AND
+    ...    Setup Telemetry Configuration    ${ENABLE_TELEMETRY_WITH_PROXY}    ${PULL_SECRET_METRICS}
+
+    Should Find Metrics In Journal Log Success    MicroShift telemetry starting, sending first metrics collection.
+    Should Find Metrics In Journal Log Success    Metrics sent successfully
+
+    [Teardown]    Run Keywords
+    ...    Remove Telemetry Configuration
+    ...    AND
+    ...    Stop Proxy Server
+
+MicroShift Fails to Report Metrics To Prometheus Server With Telemetry Disabled
+    [Documentation]    Check MicroShift is not able to send metrics to the telemetry server when it is disabled.
+    [Setup]    Run Keywords
+    ...    Start Prometheus Server
+    ...    AND
+    ...    Setup Telemetry Configuration    ${DISABLE_TELEMETRY_TO_PROMETHEUS}    ${PULL_SECRET_METRICS}
+
+    Should Find Metrics In Journal Log Success    Telemetry is disabled
+    Should Find Metrics In Journal Log Fails    Metrics sent successfully
```

Comment:
This should be redundant in the presence of previous line and the prometheus checks below.

Reply:
I agree, although I prefer to keep this line because it's the only way we can be sure metrics are not sent when it shouldn't be sent.